src/OpenProtocolInterpreter/Tool/Mid0042.cs

```diff
@@ -1,17 +1,88 @@
-﻿namespace OpenProtocolInterpreter.Tool
+﻿using OpenProtocolInterpreter.Converters;
+using System.Collections.Generic;
+
+namespace OpenProtocolInterpreter.Tool
 {
     /// <summary>
     /// Disable tool
     /// <para>Message sent by: Integrator</para>
-    /// <para>Answer: <see cref="Communication.Mid0005"/> Command accepted</para>
+    /// <para>Answer: <see cref="Communication.Mid0005"/>Command accepted</para>
+    /// <para>Answer: <see cref="Communication.Mid0004"/>Command error, with codes:
+    /// <list type="table">
+    /// <listheader>
+    /// <term>Code</term>
+    /// <description>Error description</description>
+    /// </listheader>
+    /// <item>
+    /// <term>29</term>
+    /// <description>Tool does not exist</description>
+    /// </item>
+    /// <item>
+    /// <term>36</term>
+    /// <description>Lock type not supported</description>
+    /// </item>
+    /// </list>
+    /// </para>
     /// </summary>
     public class Mid0042 : Mid, ITool, IIntegrator
     {
-        private const int LAST_REVISION = 1;
+        private readonly IValueConverter<int> _intConverter;
+        private const int LAST_REVISION = 2;
         public const int MID = 42;
 
+        public int ToolNumber
+        {
+            get => GetField(2, (int)DataFields.TOOL_NUMBER).GetValue(_intConverter.Convert);
+            private set => GetField(2, (int)DataFields.TOOL_NUMBER).SetValue(_intConverter.Convert, value);
+        }
+
+        public DisableToolType DisableType
+        {
+            get => (DisableToolType)GetField(2, (int)DataFields.DISABLE_TYPE).GetValue(_intConverter.Convert);
+            set => GetField(2, (int)DataFields.DISABLE_TYPE).SetValue(_intConverter.Convert, (int)value);
+        }
+
         public Mid0042() : base(MID, LAST_REVISION)
```

Comment:
change default constructor to call `this(LAST_REVISION)`, otherwise it will result in `_intConverter` to be null.